File: backend/src/main/java/com/silyosbekov/chessmate/dto/GameDto.java
package com.silyosbekov.chessmate.dto;

import com.silyosbekov.chessmate.model.PlayerColor;
import java.time.Instant;
import java.util.UUID;

public record GameDto(
        UUID id,
        UUID hostPlayerId,
        String hostPlayerUsername,
        PlayerColor hostPlayerColor,
        int hostPlayerElo,
        UUID whitePlayerId,
        String whitePlayerUsername,
        int whitePlayerElo,
        UUID blackPlayerId,
        String blackPlayerUsername,
        int blackPlayerElo,
        PlayerColor winnerPlayer,
        String status,
        PlayerColor currentTurn,
        boolean isRanked,
        boolean isTimerEnabled,
        String pgn,
        Instant createdDate)
{
}

File: frontend/projects/chessmate-app/src/app/core/models/game/gameDto.ts
import {GameStatus} from './gameStatus';

export interface GameDto {
  id: string;
  whitePlayerId?: string;
  whitePlayerUsername?: string;
  whitePlayerElo?: number;
  blackPlayerId?: string;
  blackPlayerUsername?: string;
  blackPlayerElo?: number;
  winnerPlayerId?: string;
  status: GameStatus;
  currentTurnPlayerId?: string;
  isRanked: boolean;
  isTimerEnabled: boolean;
  pgn: string;
  createdDate: string;
}
